raiden/tests/utils/transfer.py

```diff
@@ -171,13 +183,15 @@ def _transfer_unlocked(
         fee=fee,
     )
 
+    watch_unlock_failures = gevent.spawn(watch_for_unlock_failures, initiator_app, target_app)
     with Timeout(seconds=timeout):
         wait_for_unlock.get()
         msg = (
             f"transfer from {to_checksum_address(initiator_app.raiden.address)} "
             f"to {to_checksum_address(target_app.raiden.address)} failed."
         )
         assert payment_status.payment_done.get(), msg
+    gevent.kill(watch_unlock_failures)
```

Comment:
This is not good, if the assert fails the task is not killed and the test will be flaky. please make it more reliable. also the other tests are not using this and are still relying on the `try-finally`

Reply:
I turned it into a context manager.